src/runtime-rs/crates/runtimes/virt_container/src/sandbox.rs

```diff
@@ -117,6 +121,38 @@ impl VirtSandbox {
 
         Ok(resource_configs)
     }
+
+    async fn get_guest_details(&self) -> Result<()> {
+        let guest_details = self
+            .agent
+            .get_guest_details(GetGuestDetailsRequest {
+                mem_block_size: true,
+                mem_hotplug_probe: true,
+            })
+            .await
+            .context("failed to get guest details")?;
+        // set memory block size
+        self.hypervisor
```

Comment:
This function is a little bit fuzzy. Why getting guest memory details will set the corresponding fields?

Reply:
we use this value to calculate the memory hotplug size for ACPI hotplug, this operation is to store this value

Reply:
I am not sure about the detail of ACPI hotplug. But if I were implementing that, I would let this return `Result<GuestDetails>` and use that to further make progress.

If the case above is possible, I suggest you take that approach to implement it. But if it is not feasible or less appropriate from your view, you can ignore my suggestion :=D

Reply:
I changed the function name, maybe it looks better :)